src/Akka.Persistence.Sql/Query/Dao/BaseByteReadArrayJournalDao.cs

```diff
@@ -82,53 +84,57 @@ public abstract class BaseByteReadArrayJournalDao : BaseJournalDaoWithReadMessag
 
                 TagMode.TagTable => AsyncSource<JournalRow>
                     .FromEnumerable(
-                        new { separator, tag, offset, maxOffset, maxTake, _connectionFactory = ConnectionFactory },
-                        async input =>
+                        new { inst=this, args= new QueryArgs(offset,maxOffset,maxTake,tag,TagMode.TagTable)},
+                        static async input =>
                         {
-                            return await input._connectionFactory.ExecuteWithTransactionAsync(
-                                ReadIsolationLevel,
-                                ShutdownToken,
-                                async (connection, token) =>
+                            var inst = input.inst;
+                            return await inst.ConnectionFactory.ExecuteWithTransactionAsync(
+                                input.args,
+                                inst.ReadIsolationLevel,
+                                inst.ShutdownToken,
+                                static async (connection, token,txInput) =>
                                 {
-                                    var journalTable = connection.GetTable<JournalRow>();
-                                    var tagTable = connection.GetTable<JournalTagRow>();
-
-                                    var query =
-                                        from r in journalTable
-                                        from lp in tagTable.Where(jtr => jtr.OrderingId == r.Ordering).DefaultIfEmpty()
-                                        where lp.OrderingId > input.offset &&
-                                              lp.OrderingId <= input.maxOffset &&
-                                              !r.Deleted &&
-                                              lp.TagValue == input.tag
-                                        orderby r.Ordering
-                                        select r;
-
+                                    var query = connection.GetTable<JournalRow>()
+                                        .Where(r => r.Deleted == false)
+                                        .Join(
+                                            connection.GetTable<JournalTagRow>()
+                                                .Where(
+                                                    jtr =>
+                                                        jtr.OrderingId > txInput.Offset
+                                                        && jtr.OrderingId <= txInput.MaxOffset
+                                                        && jtr.TagValue == txInput.Tag),
+                                            SqlJoinType.Left,
+                                            (jr, jtr) => (jr.Ordering == jtr.OrderingId),
+                                            (jr, jtr) => jr)
+                                        .OrderBy(r => r.Ordering)
+                                        .Take(txInput.Max);
```

Comment:
So, We didn't have a take here.

However, AFAIR, the LINQ query syntax tends to be more finicky with captures/closures than (_carefully_) plotted out L2Db Method syntax. However, if the lambda syntax somehow is producing something needed for a specific DB, we can go back to it.